api/src/Repository/DepartmentRepository.php

```diff
@@ -18,45 +20,52 @@ public function __construct($db)
     }
 
 
-    public function insert(/*.mixed.*/$data): void
+    public function insert(/*.mixed.*/$data): array
     {
-        throw new Exception(__CLASS__.": Method '__FUNCTION__' not implemented");
+        $insert = Insert::new($this->db);
+        $insert->into('Departments')
+            ->column('Name', $data["Name"]);
+
+        if (array_key_exists("ParentID", $data)) {
+            $insert->column('ParentDepartmentID', $data["ParentID"]);
+        } else {
+            $minDeptId = Select::new($this->db)
+                ->from("Departments")
+                ->columns("MIN(DepartmentID) as DepartmentID")
+                ->fetchOne();
+            $insert->column('ParentDepartmentID', $minDeptId["DepartmentID"]);
+        }
+
+        if (array_key_exists("FallbackID", $data)) {
+            $insert->column('FallbackID', $data["FallbackID"]);
+        } else {
+            $insert->set('FallbackID', null);
+        }
+
+        $insert->perform();
+        $lastInsertId = $insert->getLastInsertId();
+
+        $result = [
+            "DepartmentID" => $lastInsertId
+        ];
```

Comment:
Yeah - no different here. Except that rather than doing this at the repository or service layer, the controller can make that decision instead.

I'll update to just return an `int` instead of an `array` here.